tests/tools/oecertdump/host/host.cpp

```diff
@@ -0,0 +1,466 @@
+#include <openenclave/internal/sgxcertextensions.h>
+#include <openenclave/internal/tests.h>
+#include <openssl/bio.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "oecertdump_u.h"
+
+#if defined(__linux__)
+#include <dlfcn.h>
+#endif
+
+#include "../../../../common/sgx/quote.h"
+#include "../../../../common/sgx/revocation.h"
+
+#include "../../../../host/sgx/sgxquoteprovider.h"
+
+#ifdef OE_LINK_SGX_DCAP_QL
+
+#define INPUT_PARAM_OPTION_OUT_FILE "--out"
+#define INPUT_PARAM_USAGE "--help"
+#define DEFAULT_OUTPUTFILE "oecertdump_out.log"
+
+// Structure to store input parameters
+//
+typedef struct _input_params
+{
+    const char* enclave_filename;
+    const char* out_filename;
+} input_params_t;
+
+static input_params_t _params;
+static FILE* log_file = NULL;
+
+void log(const char* fmt, ...)
+{
+    char message[4096];
+    va_list args;
+    va_start(args, fmt);
+    vsnprintf(message, sizeof(message), fmt, args);
+    va_end(args);
+
+    // ensure buf is always null-terminated
+    message[sizeof(message) - 1] = 0;
+
+    if (log_file)
+    {
+        fprintf(log_file, "%s", message);
+    }
+    else
+    {
+        printf("%s", message);
+    }
+}
+
+// This is the identity validation callback. An TLS connecting party (client or
+// server) can verify the passed in "identity" information to decide whether to
+// accept an connection request
+oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    log("enclave_identity_verifier is called with parsed report:\n");
+
+    // Check the enclave's security version
+    log("identity.security_version = %d\n", identity->security_version);
+    if (identity->security_version < 1)
+    {
+        log("identity.security_version check failed (%d)\n",
+            identity->security_version);
+        goto done;
+    }
+
+    // Dump an enclave's unique ID, signer ID and Product ID. They are
+    // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves.  In a real scenario,
+    // custom id checking should be done here
+    log("identity->signer_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        log("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    log("\nidentity->signer_id :\n");
+    for (int i = 0; i < OE_SIGNER_ID_SIZE; i++)
+        log("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    log("\nidentity->product_id :\n");
+    for (int i = 0; i < OE_PRODUCT_ID_SIZE; i++)
+        log("0x%0x ", (uint8_t)identity->product_id[i]);
+    log("\n");
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+// Azure dcap client log callback to this function.
+void oecertdump_quote_provider_log(
+    sgx_ql_log_level_t level,
+    const char* message)
+{
+    const char* level_string = level == 0 ? "ERROR" : "INFO";
+
+    log("[%s]: %s\n", level_string, message);
+}
+
+// Set Azure dcap calient log callback
```

Comment:
Why is this Azure specific?

Reply:
Should apply to all DCAP client. Removed "Azure".